src/model.ts

```diff
@@ -914,6 +924,7 @@ export class GitExtension implements IGitExtension, IDisposable {
   private _readyPromise: Promise<void> = Promise.resolve();
   private _pendingReadyPromise = 0;
   private _poll: Poll;
+  private _refreshInterval = 3000; // ms
```

Comment:
Related to above comment about model/settings sync, have you inspected `_refreshInterval` in devtools and made sure that it's in sync with the value in user settings under various circumstances (start up, page refresh, whatever other weird situation you can think of, etc)?

Reply:
With my recent change, the refresh interval, when configured in the user settings, persisted during both start-up and page refresh. In which case, I believe these cases are now covered.